pandas/core/dtypes/cast.py

```diff
@@ -1162,3 +1162,23 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
         subarr.fill(value)
 
     return subarr
+
+
+def cast_to_1d_array(values, dtype='object'):
+    """
+    Transform any list-like object in a 1-dimensional numpy array.
+
+    Parameters
+    ----------
+    values : any iterable which has a len()
+    dtype : dtype, default 'object'
+
+    Returns
+    -------
+    1-dimensional numpy array
+    """
+    # numpy will try to interpret nested lists as further dimensions, hence
+    # making a 1D array that contains list-likes is a bit tricky:
```

Comment:
can you add some tests for this, e.g. cases would be iterables, including a list of strings with one or more np.nan elements, and throw in some Nones